shell/src/main/java/alluxio/cli/AbstractCommand.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * The base class for all the {@link Command} classes.
+ */
+@ThreadSafe
+public abstract class AbstractCommand implements Command {
+
+  protected AbstractCommand() {}
+
+  /**
+   * Checks if the arguments are valid.
+   *
+   * @param args the arguments for the command, excluding the command name and options
+   * @return whether the args are valid
+   */
+  protected boolean validateArgs(String... args) {
+    boolean valid = args.length == getNumOfArgs();
+    if (!valid) {
+      System.out.println(getCommandName() + " takes " + getNumOfArgs() + " arguments, " + " not "
+          + args.length + "\n");
+    }
+    return valid;
+  }
+
+  /**
+   * Gets the expected number of arguments of the command.
+   *
+   * @return the number of arguments
+   */
+  protected abstract int getNumOfArgs();
+
+  @Override
+  public Options getOptions() {
+    return new Options();
+  }
+
+  @Override
+  public CommandLine parseAndValidateArgs(String... args) {
```

Comment:
Why do we want to return `null` here instead of just throwing an exception? It seems like this is not an optional method.